### PR TITLE
feat(#major); sushiswap v3; add multiple networks

### PR DESCRIPTION
**Pull request by OlaStenberg**
Added sushi v3 configuration

Deployed one of them here:
https://thegraph.com/hosted-service/subgraph/olastenberg/sushiswap-v3-ethereum

### CODE REVIEW

File: package-lock.json
Comment (dmelotik) [inline package-lock.json:1]: Could you undo the changes to this file?

**Comment by dmelotik** [comment]
Hey @OlaStenberg thank you for the PR! We will begin reviewing, I left 1 small comment to start. As a note, you can ignore the linting issues. We are in the process of changing that.

File: deployment/deployment.json [hunk 1]
```diff
@@ -3721,6 +3721,36 @@
       }
     }
   },
+  "sushiswap-v3": {
+    "schema": "dex-amm",
+    "base": "uniswap-v3-forks",
+    "protocol": "sushiswap-v3",
+    "project": "sushiswap",
+    "deployments": {
+      "sushiswap-v3-ethereum": {
```
Comment (OlaStenberg) [inline deployment/deployment.json:3730]: I just realised I didn't add all of them here, assume I should? @dmelotik

File: subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-arbitrum/configurations.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    log.warning("getProtocolFeeRatio is not implemented: {}", [fee.toString()]);
+    return BIGDECIMAL_ZERO;
+  }
+  getProtocolFeeProportion(protocolFee: BigInt): BigDecimal {
+    return BIGDECIMAL_ONE.div(protocolFee.toBigDecimal());
+  }
+  getRewardIntervalType(): string {
+    return RewardIntervalType.NONE;
+  }
+  getReferenceToken(): Bytes {
```
Comment (steegecs) [inline subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-arbitrum/configurations.ts:51]: Would you leave a comment next to the reference token indicating which token it is with the symbol like the others?

File: deployment/deployment.json (hunk 1 above)
Comment (steegecs) [reply deployment/deployment.json:3730]: Hi Ola! Yes, please add all of the networks to the deployment.json.

File: subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-moonriver/configurations.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  }
+  getFactoryAddress(): Bytes {
+    return Bytes.fromHexString("0x2f255d3f3c0a3726c6c99e74566c4b18e36e3ce6");
+  }
+  getFactoryContract(): Factory {
+    return Factory.bind(
+      Address.fromString("0x2f255d3f3c0a3726c6c99e74566c4b18e36e3ce6")
+    );
+  }
+  getProtocolFeeOnOff(): string {
```
Comment (steegecs) [inline subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-moonriver/configurations.ts:38]: Is the protocol fee currently on? You might need to flip these on if so.

File: subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/src/core.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { BigInt } from "@graphprotocol/graph-ts";
+import { updateProtocolFees } from "../../../src/common/entities/pool";
+import { SetFeeProtocol } from "../../../generated/templates/Pool/Pool";
+
+// Update the fees colected by the protocol.
+export function handleSetFeeProtocol(event: SetFeeProtocol): void {
```
Comment (steegecs) [inline subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/src/core.ts:6]: Is the protocol fee on?

**Review by steegecs — COMMENTED** [review]
Nice job! Just a few comments.

File: deployment/deployment.json (hunk 1 above)
Comment (OlaStenberg) [reply deployment/deployment.json:3730]: https://github.com/messari/subgraphs/pull/2151/commits/a282811d6d3f0d347f2b9dad66de08189230888f

File: subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-arbitrum/configurations.ts (hunk 2 above)
Comment (OlaStenberg) [reply subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-arbitrum/configurations.ts:51]: https://github.com/messari/subgraphs/pull/2151/commits/fa66f9ce0e791be71b2d26b1c56d28c7512b0d38

File: subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-moonriver/configurations.ts (hunk 3 above)
Comment (OlaStenberg) [reply subgraphs/uniswap-v3-forks/protocols/sushiswap-v3/config/deployments/sushiswap-v3-moonriver/configurations.ts:38]: https://github.com/messari/subgraphs/pull/2151/commits/6e49600a46e33ab691a00d2a15f58f8053113ce3

**Review by steegecs — APPROVED** [review]
Looks good! Thanks.
